pkg/planner/core/point_get_plan.go

```diff
@@ -996,7 +1022,13 @@ func TryFastPlan(ctx base.PlanContext, node *resolve.NodeW) (p base.Plan) {
 			return
 		}
 		if fp := tryPointGetPlan(ctx, x, node.GetResolveContext(), isForUpdateReadSelectLock(x.LockInfo)); fp != nil {
-			if checkFastPlanPrivilege(ctx, fp.dbName, fp.TblInfo.Name.L, mysql.SelectPriv) != nil {
+			visitInfos := getVisitInfo(userName, hostName, fp.dbName, fp.TblInfo.Name.L, fp.outputNames, fp.unfoldFromWildCard, fp.colsInWhereClause)
+			failpoint.Inject("point-get-visit-info", func(val failpoint.Value) {
+				if val.(bool) {
+					pointGetVisitInfo4Test = visitInfos
+				}
+			})
+			if checkFastPlanPrivilege(ctx, visitInfos...) != nil {
```

Comment:
I've noticed a lot of repetition: getVisitInfo, failpoint.Inject("point-get-visit-info"), and finally checkFastPlanPrivilege. Can we encapsulate these into a function?